lib/chef_zero/server.rb

```diff
@@ -164,10 +164,10 @@ def port
     #
     def url
       sch = @options[:ssl] ? 'https' : 'http'
-      @url ||= if @options[:host].include?(':')
-                 URI("#{sch}://[#{@options[:host]}]:#{port}").to_s
+      @url ||= if @options[:host].first.include?(':')
+                 URI("#{sch}://[#{@options[:host].first}]:#{port}").to_s
                else
-                 URI("#{sch}://#{@options[:host]}:#{port}").to_s
+                 URI("#{sch}://#{@options[:host].first}:#{port}").to_s
```

Comment:
Dude, this would never worked as expected even before this. Consider the most obvious case, '0.0.0.0' or '::' - either way you're getting a bullshit URL back. This has never been a useful URL, and as you pointed out, we ignore it anyway.